ur_controllers/src/force_torque_sensor_controller.cpp

```diff
@@ -0,0 +1,180 @@
+//
+// Created by livanov93 on 12/12/20.
+//
+
+#include "ur_controllers/force_torque_sensor_controller.h"
+
+namespace ur_controllers
+{
+ForceTorqueStateController::ForceTorqueStateController() : controller_interface::ControllerInterface()
+{
+}
+
+controller_interface::return_type ForceTorqueStateController::init(const std::string& controller_name)
+{
+  auto ret = ControllerInterface::init(controller_name);
+  if (ret != controller_interface::return_type::SUCCESS)
+  {
+    return ret;
+  }
+
+  return controller_interface::return_type::SUCCESS;
+}
+
+controller_interface::InterfaceConfiguration ForceTorqueStateController::command_interface_configuration() const
+{
+  controller_interface::InterfaceConfiguration config;
+  config.type = controller_interface::interface_configuration_type::NONE;
+  return config;
+}
+
+controller_interface::InterfaceConfiguration ForceTorqueStateController::state_interface_configuration() const
+{
+  controller_interface::InterfaceConfiguration config;
+  config.type = controller_interface::interface_configuration_type::INDIVIDUAL;
+  // TODO solve this hardcoded naming
+  config.names.emplace_back("tcp_fts_sensor/fx");
+  config.names.emplace_back("tcp_fts_sensor/fy");
+  config.names.emplace_back("tcp_fts_sensor/fz");
+  config.names.emplace_back("tcp_fts_sensor/tx");
+  config.names.emplace_back("tcp_fts_sensor/ty");
+  config.names.emplace_back("tcp_fts_sensor/tz");
+  return config;
+}
+
+controller_interface::return_type ur_controllers::ForceTorqueStateController::update()
+{
+  geometry_msgs::msg::Vector3 fVec;
+  geometry_msgs::msg::Vector3 tVec;
+
+  // TODO remove these hardcoded names and create better string filtering within state_interfaces_
+  for (const auto& state_interface : state_interfaces_)
```

Comment:
Optimization: it would probably be better to sort interfaces after they are assigned. This can be done by overriding `assign_interfaces` [method](https://github.com/ros-controls/ros2_control/blob/master/controller_interface/include/controller_interface/controller_interface.hpp#L79).

After that those can be used directly.